### PR TITLE
API-166: Get a single attribute group and get list of attribute groups

### PR DESCRIPTION
**Pull request by momoss**
**Description (for Contributor and Core Developer)**

Get a single attribute group
Get list of attribute groups

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | OK
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        $this->repository = $repository;
+        $this->normalizer = $normalizer;
+        $this->parameterValidator = $parameterValidator;
+        $this->paginator = $paginator;
+        $this->apiConfiguration = $apiConfiguration;
+    }
+
+    /**
+     * @param Request $request
+     * @param         $code
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: string

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            'with_count' => 'false',
+        ];
+
+        $queryParameters = array_merge($defaultParameters, $request->query->all());
+
+        $offset = $queryParameters['limit'] * ($queryParameters['page'] - 1);
+        $attributeGroups = $this->repository->searchAfterOffset([], ['code' => 'ASC'], $queryParameters['limit'], $offset);
+
+        $parameters = [
+            'query_parameters'    => $queryParameters,
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: spaces

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+	"_links": {
+		"self": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=1&limit=10&with_count=false"
+		},
+		"first": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=1&limit=10&with_count=false"
+		}
+	},
+	"current_page": 1,
+    "_embedded" : {
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php]: indentation

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+		},
+		"previous": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=1&limit=1&with_count=true"
+		},
+		"next": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=3&limit=1&with_count=true"
+		}
+	},
+    "current_page": 2,
+	"items_count": 3,
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php]: indentation

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=2&limit=100&with_count=false"
+		},
+		"first": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=1&limit=100&with_count=false"
+		},
+		"previous": {
+			"href": "http://localhost/api/rest/v1/attribute-groups?page=1&limit=100&with_count=false"
+		}
+	},
+    "current_page": 2,
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php]: indentation

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+	"_embedded" : {
+        "items" : []
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/ListAttributeGroupIntegration.php]: indentation

File: src/Pim/Bundle/ApiBundle/Resources/config/routing/attribute.yml [hunk 7]
```diff
@@ -42,3 +42,13 @@ pim_api_attribute_option_get:
     path: /attributes/{attributeCode}/options/{code}
     defaults: { _controller: pim_api.controller.attribute_option:getAction, _format: json }
     methods: [GET]
+
+pim_api_attribute_group_list:
+    path: /attribute-groups
+    defaults: { _controller: pim_api.controller.attribute_group:listAction, _format: json }
+    methods: [GET]
+
+pim_api_attribute_group_get:
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/Resources/config/routing/attribute.yml]: not in a separated file for the attribute group ?

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/GetAttributeGroupIntegration.php [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        $this->assertSame(Response::HTTP_NOT_FOUND, $content['code']);
+        $this->assertSame('Attribute group "not_found" does not exist.', $content['message']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getConfiguration()
+    {
+        return new Configuration(
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/GetAttributeGroupIntegration.php]: Yohan has merged his PR, you can rebase from master and remove `false` parameter

File: src/Pim/Component/Api/Normalizer/AttributeGroupNormalizer.php [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($group, $format = null, array $context = [])
+    {
+        $normalizedGroup = $this->stdNormalizer->normalize($group, 'standard', $context);
+
+        if (empty($normalizedGroup['labels'])) {
+            $normalizedGroup['labels'] = (object)$normalizedGroup['labels'];
```
Comment (skeleton) [inline src/Pim/Component/Api/Normalizer/AttributeGroupNormalizer.php]: add space after `(object)` please

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php (hunk 1 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: ok

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php (hunk 2 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: ok

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/GetAttributeGroupIntegration.php (hunk 8 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/GetAttributeGroupIntegration.php]: Not on master at this time

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        try {
+            $this->parameterValidator->validate($request->query->all());
+        } catch (PaginationParametersException $e) {
+            throw new UnprocessableEntityHttpException($e->getMessage(), $e);
+        }
+
+        $defaultParameters = [
+            'page'       => 1,
+            'limit'      => $this->apiConfiguration['pagination']['limit_by_default'],
+            'with_count' => 'false',
```
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: Without quote, it interpreted as `O` or `1` in request

**Pull request merged**